odoo/addons/base/models/ir_cron.py

```diff
@@ -320,5 +320,12 @@ def try_write(self, values):
 
     @api.model
     def toggle(self, model, domain):
+        # Prevent deactivated cron jobs from being re-enabled through side effects on
+        # neutralized databases. Invalidate the parameter cache so the neutralization
+        # applies without restarting the database.
+        self.env['ir.config_parameter'].clear_caches()
```

Comment:
I would avoid this cache clearing, as it will be done systematically and would cause collateral damage in performance. In normal situations any call to `set_param()` is responsible for sending a "cache clear" signal that should take care of it[^1].
It may not hold true in the special CLI situation we have in #85205 (there's no request cycle), but then we could fix with less side-effects by adding a call to `env.registry.signal_changes()` after the `set_param`, like we do it explicitly for cron jobs, for example [^2]

[^1]: https://github.com/odoo/odoo/blob/13.0/odoo/addons/base/models/ir_config_parameter.py#L74-L109
[^2]: https://github.com/odoo/odoo/blob/13.0/odoo/addons/base/models/ir_cron.py#L114

Reply:
Yeah, I wasn't happy about the performance part, thanks for the alternative solution. I'm also stealing your technique of putting links in footnotes.